Обновление скрипта

dzharlaksl opened the pull request:
Переместил скрипт в папку src
Согласно комментариям учел работу с переменными категории и приоритетов. Расширил и изменил логику работы функции для более точной выдачи результата. Полностью переписал сам скрипт

dzharlaksl committed:
Обновление скрипта

Переместил скрипт в папку src
Согласно комментариям учел работу с переменными категории и приоритетов. РАсширил и изменил логику работы функции для более точной выдачи результата. Полностью переписал сам скрипт.

MonakhovVadim commented on src/categorize.py:
```diff
@@ -0,0 +1,82 @@
+    return results
+
+
+print(process_texts(texts_to_process))
```
Строки в теле питоновского файла вызываются всякий раз при инициализации модуля, поэтому так оставлять их плохо. По-хорошему тестирование лучше реализовывать через pytest, который сделает отдельный участник. Как временную меру можно использовать конструкцию if __name__ == '__main__': и уже в теле этого ветвления организовывать проверки. __name__ - это переменная, которая хранит имя файла, но если запускать файл напрямую командой "python categorize.py", то имя этой переменной для запускаемого файла будет '__main__'. Таким образом, код в условии будет работать только в случае, если файл запустили напрямую. Когда наш проект будет работать, то твой модуль никогда не будет запускаться напрямую, его будут использовать другие модули.

MonakhovVadim commented on src/categorize.py:
```diff
@@ -0,0 +1,82 @@
+import re
```
Импорт есть, а регулярных выражений нет.

MonakhovVadim reviewed (commented):
Про категории мы уже говорили, вероятно потребуется встреча с голосом. У нас для ветки develop настроен линтер, он пометит ошибки оформления, это надо будет тоже поправить для чистоты кода.

MonakhovVadim committed:
Refactor categories
Категории переработы, теперь это словарь из именнованных кортежей,
которые имеют поля name(0) - имя категории и key_words(1) - список
строк, содержащих ключевые слова и предложения.

MonakhovVadim committed:
Rebuild functions

Добавлена фкнкция name_category, которая возвращает имя категории по
числовому id. categorize_text теперь возвращает числовой id вместо
название категории. process_texts теперь работает со строковым элементом
вместо массива.

MonakhovVadim committed:
Refactor priorities
Приоритеты приведены к стилю категорий. А именно теперь это словарь из
кортежей. В качестве выозврата приоритетов используются числовые id.
Добавлена отдельная функция для получения имени приорите по id.

bev141 commented on src/categorize.py:
```diff
@@ -0,0 +1,322 @@
+def categorize_text(text):
+
+    for id_cat, category in categories.items():
+        if any(word in text for word in category.key_words):
```
В предыдущей версии текст к нижнему регистру сначала приводился. Теперь как есть сравнивается, так что если слово в начале предложения, то вероятно алгоритм его не найдет. Думаю лучше все привести к нижнему регистру, и текст и ключеные слова в массиве.

bev141 commented on src/categorize.py:
```diff
@@ -0,0 +1,322 @@
+    return 0
+
+
+def name_category(id):
```
По-моему на английском category_name и priority_name более правильно...

MonakhovVadim committed:
Change classifier
Изменен алгоритм классификации. Теперь текст очищается от лишних знаков
и предлогов. Отдельные слова приводятся к нормальной форме. Для
определения наиболее подходящей категории используется подобие Жаккара.

MonakhovVadim committed:
Rename functions

MonakhovVadim committed:
Change keywords
Из ключевых слов убраны ключевые фразы. Алгоритм определения приоритета
теперь использует тот же алгоритм, что и анализ категорий.

operevispa commented on src/categorize.py:
```diff
@@ -0,0 +1,354 @@
+
+# Основная функция скрипта
+def process_texts(text):
+
```
Допиши, пожалуйста, описание функции, особенно возвращаемых параметров - это важно для других членов команды, которые будут пользоваться именно этой функцией для своих задач

operevispa replied:
И поменяй, пожалуйста, название этой функции. По ее названию непонятно, что она делает и для чего нужна.

MonakhovVadim replied:
Готово

operevispa reviewed (changes requested):
Использование подобия Жаккара отличное решение для этой задачи! Лучше разве что использование полноценной модели ИИ обученной на предметной области.
Качество классификации категории существенно улучшилось. Молодец!
Пожалуйста, поменяй название "основной" функции на более говорящее и добавь описание возвращаемых параметров, чтобы другие члены команды могли без проблем использовать функцию определения категории и приоритета заявки.
Спасибо!

MonakhovVadim committed:
Add description
